dandiapi/api/models/metadata.py

```diff
@@ -19,7 +21,8 @@ def published_by(self, now: datetime.datetime):
             'wasAssociatedWith': [
                 {
                     'id': uuid4().urn,
-                    'identifier': 'RRID:SCR_017571',
+                    # TODO: handle the case where `instance_identifier` is None
+                    'identifier': settings.DANDI_SCHEMA_INSTANCE_CONFIG.instance_identifier,
```

Comment:
make it optional and don't add if `not` .

below, please add also `version` to match the version from `/api/info` -- it is not per se API version but would be useful info to know with what version of archive we did it. I think it is much better than some ad-hoc fixed version since API etc change and reflect in the version of server implementation.  May be even change name to "{instance_name} API Server", which would disambiguate instances, and also that it is not just API version but the version of the server.